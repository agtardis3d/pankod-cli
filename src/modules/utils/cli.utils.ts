// #region Local Imports
import * as suitcase from '../henchman/suitcase';
import { ICommon } from '../typings';
import { renderMessage } from '.';
// #endregion Local Imports

// TODO: bind this or what ?
export const validateProject = (project: string) => {
    if (!suitcase.hasOwnProperty(project)) {
        this.error(renderMessage.invalidProject(project));
    }
};

export const validateCommand = (
    element: ICommon.Element,
    project: ICommon.Project
) => {
    if (!suitcase[project].hasOwnProperty(element)) {
        this.error(renderMessage.invalidElement(project, element));
    }
};

export const getAllElements = () => {
    return Object.keys(Object.assign({}, ...Object.values(suitcase)));
};

export const getUsage = () => getAllElements().map(e => `add ${e}`);
