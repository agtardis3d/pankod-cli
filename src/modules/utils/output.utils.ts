// #region Global Imports
import chalk from 'chalk';
// #endregion Global Imports

// TODO: Create a better renderer
export const renderMessage = {
    invalidProject: (project: string) => {
        return `${chalk.bold(project)} is NOT a supported project!`;
    },

    invalidElement: (project: string, element: string) => {
        const [p, e] = [project, element].map(w => chalk.bold(w));

        return chalk.red(
            `${p} projects have no element called ${e} to be created!`
        );
    }
};
