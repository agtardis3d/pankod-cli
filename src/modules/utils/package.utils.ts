// #region Global Imports
import * as fs from 'fs';
// #region Global Imports

// #region Local Imports
import { IPankodConfig } from '../typings';
// #endregion Local Imports

export const getPankodConfig = (): IPankodConfig => {
    const config = JSON.parse(String(fs.readFileSync('./package.json'))) as {
        pankod: IPankodConfig;
    };

    return config.pankod;
};

export const hasPlugin = (pluginName: string): boolean => {
    const plugins: string[] = getPankodConfig().plugins;

    return plugins.includes(pluginName);
};
