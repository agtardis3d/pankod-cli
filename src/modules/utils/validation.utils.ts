
// TODO: Refactor properly
export const validateProjectSupported = (projectType: string) => {
    if (!Object.keys(choices).includes(projectType)) {
        this.error(`The project ${projectType} isn't supported.`);
    }
};

export const validateCommand = (element: string, projectType: string) => {
    if (element && !choices[projectType].includes(element)) {
        this.error(`Command "${element}" isn't supported by ${projectType} project.`);
    }
};
