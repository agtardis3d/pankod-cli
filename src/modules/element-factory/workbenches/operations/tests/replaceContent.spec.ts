// #region Global Imports
import * as fs from 'fs';
import * as path from 'path';
// #endregion Global Imports

// #region Local Imports
import { replaceContent } from '../replaceContent.operation';
import { failsafe } from '../failsafe.operation';
import { operations } from '../../../../utils';
// #endregion Local Imports

describe(operations, () => {
    describe('replaceContent.operation', () => {
        it('should replace regexp matchings with passed contents.', () => {
            const sample = './__temp__/replaceContent/target.ts';

            const sampleContent = '// ! I was just sitting here in silence...';

            failsafe(sample);
            fs.writeFileSync(path.resolve(sample), sampleContent);

            const params = {
                fileDir: sample,
                filetoUpdate: sampleContent,
                regexKey: /\/\/ ! I was just sitting here in silence.../g,
                getFileContent: () => `${sampleContent}\nThen piston dropped!`,
                message: 'replaceContent.operation works!'
            };

            replaceContent(params);

            expect(fs.readFileSync(path.resolve(sample), 'utf8')).toMatchSnapshot();
        });
    });
});
