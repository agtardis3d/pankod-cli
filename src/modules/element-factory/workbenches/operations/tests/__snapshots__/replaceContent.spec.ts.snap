// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[44m[1m[operations][22m[49m -- [3mfs manipulations[23m replaceContent.operation should replace regexp matchings with passed contents. 1`] = `
"// ! I was just sitting here in silence...
Then piston dropped!"
`;
